### ISSUE TITLE
[Question] Documenting "inherited" method

### ISSUE THREAD

**Issue by Eneroth3** [lsegal/yard#1314]
Ruby has a special `inherited` method that gets called when a class is inherited. How should I document this in Yard? Since this is a hook defined by Ruby and merely implemented by me, I have a feeling I shouldn't document what it does and when, but reference it somewhere else.

Does this look right to you?

```ruby
# Example parent class.
class Parent
  # @api
  # @see https://ruby-doc.org/core-2.6.3/Class.html#method-i-inherited
  def self.inherited(subclass)
    puts "#{self} inhertied by #{subclass}"
  end
end

# Example child class.
class Child < Parent; end
```

In a broader perspective the question applies to all Ruby hooks/callbacks like extended, prepended and inherited, but also has some resemblance to constructors like initialize, intialize_copy, initialize_clone and initialize_dup in how the method is invoked by Ruby itself.

Thanks

**Comment by MSP-Greg** [comment]
One thing to remember is that 'hook/callback' methods can be called by any code, so one could say that they are no different than other methods.

One could use `@!group` directives to indicate that they are different...

**Comment by Eneroth3** [comment]
It's true the hooks are at a low level methods that can be invoked like any other method, but on a conceptual level they are wildly different. I'm not the one dictating what this method does, what parameters it takes, what it returns etc. I'm merely obeying someone else's API here.

**Comment by MSP-Greg** [comment]
I don't disagree.  You've mentioned some of Ruby Core's hooks/callbacks, but they also exist with RubyGems, OpenSSL, etc.  Gems separate from Ruby also use them.

I suppose a tag like @callback or @hook might be helpful, but one could also use @note, or create their own tag like @note...

**Comment by lsegal** [comment]
> Does this look right to you?

This seems overdocumented to me. You certainly don't need to URL link to core documentation. When YARD generates docs with enough global context, it can make the association between callbacks like `Class#inherited` on its own. 

There have been discussions about modeling Ruby core documentation and making it available as a base docset that can be included when core methods are referenced or inherited, but there's a bunch of complexity that this brings and so it hasn't (yet) been implemented. You could do this on a small scale with a plugin or custom hook as noted above, but linking all the way to the docs seems excessive.

As a sidenote, `@api` takes a parameter. My general suggestion here would be to mark this API as "internal" (`@api internal`) and have your _default_ docs omit those APIs (using `--query "@api.text != 'internal'"`) since users of your external APIs don't need to see this. Implementors can generate docs for internals with a different command or you can forego this step if you have a small enough maintainer team and you are all comfortable looking directly at source for these things.

TLDR: the following is sufficient:

```ruby
# @api internal
def self.inherited(subclass) ... end
```
